fix: correct MockFileInfo  and MockDirectoryInfo for non existing files/directories

siprbaum opened the pull request:
fixes #829 
closes #830

fgreinacher commented:
@siprbaum I agree that we should make `MockFileData.NullObject` internal to prevent misunderstandings. The changes look great! Please do the [version bump](https://github.com/TestableIO/System.IO.Abstractions/blob/main/version.json) as part of this PR and rebase against `main`.

siprbaum commented:
Rebased and added a commit on top which inreases the major version number.
Please tell if this should be squashed into the existing commit, e.g. https://github.com/TestableIO/System.IO.Abstractions/pull/834/commits/c927c6b99a0a1e16b8071b005dd9f97b35df84fe

It is not clear if increasing the version per PR is fine or if we want to do it per commit, as I believe keeping the history
of this series seems like a worthwhile goal, especially as there is a lot of information in the commits!
